README.md

```diff
@@ -1,85 +1,123 @@
+and [IERC721WithData](samples/solidity/contracts/IERC721WithData.sol) will provide the most complete
+FireFly functionality by allowing FireFly transactions and messages to be pinned to the blockchain
+alongside token operations. The sample [ERC20WithData](samples/solidity/contracts/ERC20WithData.sol)
+and [ERC721WithData](samples/solidity/contracts/ERC721WithData.sol) contracts may be used to
+get up and running with simple token support, and may provide a starting point for developing
+production contracts that can be used with this connector.
 
-```
-config: {
-  address?: string, // Address of a pre-deployed contract to index
-  blockNumber?: string, // Starting block to begin indexing (only valid with "address")
-}
-```
+### Solidity Interface Support
 
-**POST /approval**
+In the absence of being provided with ABI details, the token connector will attempt to guess the contract
+ABI in use. It does this by using ERC165 `supportsInterface()` to query the contract's support for `IERC20WithData`
+or `IERC721WithData`, as defined in this repository. If the query succeeds, the connector will leverage
+the methods on that interface to perform token operations. Therefore it is possible to use these
+contracts without the extra step of teaching FireFly about the contract interface first.
 
-```
-config: {
-  allowance?: string, // Number of tokens operator is allowed to manage (only valid with ERC20)
-  tokenIndex?: string, // A specific NFT operator is allowed to manage (only valid with ERC721)
-}
-```
+### Fallback Functionality (not recommended)
+
+If neither of the above procedures is followed for a given contract, the connector will fall back to assuming
+that the ABI looks like [ERC20NoData.json](src/abi/ERC20NoData.json) or
+[ERC721NoData.json](src/abi/ERC721NoData.json), which are based on common OpenZeppelin patterns. This
+behavior can also be tweaked to assume [ERC20NoDataOld.json](src/abi/ERC20NoDataOld.json) or
+[ERC721NoDataOld.json](src/abi/ERC721NoDataOld.json) by setting `USE_LEGACY_ERC20_SAMPLE=true` or
+`USE_LEGACY_ERC721_SAMPLE=true` in the connector environment (these sample ABIs were provided in an older version
+of this repository but are now deprecated). However, relying on this fallback functionality may be unreliable
+and is not recommended.
+
+## API Extensions
+
+The APIs of this connector conform to the FireFly fftokens standard, and are designed to be called by
+FireFly. They should generally not be called directly by anything other than FireFly.
+
+Below are some of the specific considerations and extra requirements enforced by this connector on
+top of the fftokens standard.
+
+### `/createpool`
+
+If `config.address` is specified, the connector will index the token contract at the specified address
+(must be an ERC20 contract if `type` is `fungible`, or an ERC721 contract if `type` is `nonfungible`).
+`config.blockNumber` may also be supplied to begin indexing from a specific block. Any `name` provided
```

Comment:
Good question. Looks like it is very explicitly a default of `0`. I can add it to the README, but it's worth asking if that's the behavior we want? I guess it could be disruptive to change the default at this point though.

Reply:
Both newest and oldest are going to be the "wrong" default choice for different circumstances. An explicit block number is almost always going to be the "correct" way to do it in the real world. I think for development purposes, `0` is appropriate on small test chains.

Reply:
I am in favor of simply documenting the existing default.

Reply:
I think documenting the existing default makes sense (`latest` is almost never the right answer for tokens, and apart from `0` it doesn't seem like there are good alternatives as a default)

Reply:
Yea agreed. `latest` is probably just setting you up for totally incorrect balance reporting. Will document the default.